### PR TITLE
Multiple delete fix

### PR DESCRIPTION
**Pull request by ohadkoren**
- Update the table once the client delete rows from all-experiments table.
- Update the counter of experiments accordingly.
- Make sure we can do this delete multiple times.
- Passed integration with real server.

### CODE REVIEW

File: portal/src/app/components/all-experiments/Experiments.tsx [hunk 1]
```diff
@@ -26,24 +26,41 @@ const DuplicateAriaLabel: string = ALL_EXPERIMENTS_TABLE_EN.TABLE_COLUMNS.LINKS.
...
   const { testSuites, status }: IUseExperimentsData = useExperimentsData();
+  const navigate = useNavigate();
   const [openDeleteModal, setOpenDeleteModal] = useState<boolean>(false);
   const [checkedRows, setCheckedRows] = useState<Record<number, boolean>>({});
-  const experimentsData = useMemo(() => (testSuites ? parseExperimentsData(testSuites): []), [testSuites]);
-  const navigate = useNavigate();
-
+  const [experimentsData, setExperimentsData] = useState<ExperimentData[]>([]);
   
+  const [deleteHandled, setDeleteHandled] = useState(false);
```
Comment (adibar121) [inline portal/src/app/components/all-experiments/Experiments.tsx:34]: for what you need this state?

Comment (ohadkoren) [reply portal/src/app/components/all-experiments/Experiments.tsx:34]: We need it in order to run the useEffect only once and not infinite times

Comment (adibar121) [reply portal/src/app/components/all-experiments/Experiments.tsx:34]: you can handle all values that you want to delete and then is the deleted success you can filter all this values from `experimentsData` array

  const [deletedData, setDeletedData] = useState<ExperimentData[]>([]);
and use new `deletedData` state to filter values from the array

and you can remove the deleteHandled flag state

Comment (ohadkoren) [reply portal/src/app/components/all-experiments/Experiments.tsx:34]: You're right, that's a better writing.
Fixed :)

**Review by adibar121 — APPROVED** [review]

**Pull request merged**